### PR TITLE
Add tests for SMB protocol NULL and empty DACL behavior

### PR DESCRIPTION
**Pull request by anodos325**

### CODE REVIEW

**Review by mgrimesix — APPROVED** [review]

File: tests/api2/test_smb_null_empty_dacl.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+def test_empty_dacl_set(unprivileged_user_fixture, share):
+    """ verify that setting empty DACL results in expected ZFS ACL """
+    with smb_connection(
+        share=share['share']['name'],
+        username=unprivileged_user_fixture.username,
+        password=unprivileged_user_fixture.password,
+    ) as c:
+        fh = c.create_file('test_null_dacl', 'w')
```
Comment (bmeagherix) [inline tests/api2/test_smb_null_empty_dacl.py:106]: Should the filename here be `test_empty_dacl `?

File: tests/api2/test_smb_null_empty_dacl.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        current_sd = c.get_sd(fh, security.SECINFO_OWNER | security.SECINFO_GROUP)
+        current_sd.dacl = security.acl()
+        c.set_sd(fh, current_sd, security.SECINFO_OWNER | security.SECINFO_GROUP | security.SECINFO_DACL)
+
+        new_sd = c.get_sd(fh, security.SECINFO_OWNER | security.SECINFO_GROUP)
+        assert new_sd.dacl.num_aces == 0
```
Comment (bmeagherix) [inline tests/api2/test_smb_null_empty_dacl.py:112]: If this _**is**_ an empty DACL test, then how would we have permissions to read this thru SMB?

Comment (anodos325) [reply tests/api2/test_smb_null_empty_dacl.py]: This is why the next test verifies that non-owner gets access denied.

**Review by bmeagherix — APPROVED** [review]
